esp_isotp/esp_isotp.c

```diff
@@ -0,0 +1,237 @@
+
+    // Allocate memory for handle
+    esp_isotp_handle_t handle = calloc(1, sizeof(struct esp_isotp_link_t));
+    ESP_GOTO_ON_FALSE(handle != NULL, ESP_ERR_NO_MEM, err, TAG, "Failed to allocate memory for ISO-TP link");
+
+    // Create RX queue
+    handle->rx_queue = xQueueCreate(config->rx_queue_size, sizeof(isotp_rx_queue_item_t));
+    ESP_GOTO_ON_FALSE(handle->rx_queue != NULL, ESP_ERR_NO_MEM, err_handle, TAG, "Failed to create RX queue");
+
+    handle->twai_node = twai_node;
+
+    handle->tx_buffer = calloc(config->tx_buffer_size, sizeof(uint8_t));
+    ESP_GOTO_ON_FALSE(handle->tx_buffer != NULL, ESP_ERR_NO_MEM, err_tx_buf, TAG, "Failed to allocate memory for TX buffer");
+
+    handle->rx_buffer = calloc(config->rx_buffer_size, sizeof(uint8_t));
+    ESP_GOTO_ON_FALSE(handle->rx_buffer != NULL, ESP_ERR_NO_MEM, err_rx_buf, TAG, "Failed to allocate memory for RX buffer");
+
+    // Initialize ISO-TP link with static buffers
+    isotp_init_link(&handle->link, config->tx_id, handle->tx_buffer,
+                    config->tx_buffer_size, handle->rx_buffer, config->rx_buffer_size,
+                    handle->twai_node);
+    handle->link.receive_arbitration_id = config->rx_id;
+
+    // Register TWAI callback
+    twai_event_callbacks_t cbs = {
+        .on_rx_done = esp_isotp_rx_callback,
+    };
+    ret = twai_node_register_event_callbacks(handle->twai_node, &cbs, handle);
+    ESP_GOTO_ON_ERROR(ret, err_callbacks, TAG, "Failed to register event callbacks: %s", esp_err_to_name(ret));
+
+    // Enable TWAI node
+    ret = twai_node_enable(handle->twai_node);
+    ESP_GOTO_ON_ERROR(ret, err_enable, TAG, "Failed to enable TWAI node: %s", esp_err_to_name(ret));
+
+    ESP_LOGI(TAG, "New ISO-TP link created, TX: 0x%lX, RX: 0x%lX", config->tx_id, config->rx_id);
+    *out_handle = handle;
+    return ESP_OK;
+
+err_enable:
+err_callbacks:
+    free(handle->rx_buffer);
+err_rx_buf:
+    free(handle->tx_buffer);
+err_tx_buf:
+    if (handle->rx_queue) {
+        vQueueDelete(handle->rx_queue);
+    }
+err_handle:
+    free(handle);
+err:
+    if (out_handle != NULL) {
+        *out_handle = NULL;
+    }
+    return ret;
+}
+
+esp_err_t esp_isotp_poll(esp_isotp_handle_t handle)
+{
+    ESP_RETURN_ON_FALSE(handle != NULL, ESP_ERR_INVALID_ARG, TAG, "Invalid handle");
+
+    // Process received queue data
+    isotp_rx_queue_item_t rx_item;
+    while (xQueueReceive(handle->rx_queue, &rx_item, 0) == pdTRUE) {
+        isotp_on_can_message(&handle->link, rx_item.data, rx_item.data_len);
+    }
+
+    // Run ISO-TP state machine
+    isotp_poll(&handle->link);
+
+    return ESP_OK;
+}
+
+esp_err_t esp_isotp_send(esp_isotp_handle_t handle, const uint8_t *data, uint16_t size)
+{
+    ESP_RETURN_ON_FALSE(handle != NULL, ESP_ERR_INVALID_ARG, TAG, "Invalid handle");
+    ESP_RETURN_ON_FALSE(data != NULL, ESP_ERR_INVALID_ARG, TAG, "Invalid data");
+
+    int ret = isotp_send(&handle->link, data, size);
```

Comment:
Is the `esp_isotp` restricted to single threaded calling contexts? For example, if an IDF application spins up multiple tasks, and those tasks both call `esp_isotp_send()` simultaneously, do we need to make sure that access to the `handle->link` data structure is thread safe?